src/app/_hooks/useCategory.ts

```diff
@@ -1,70 +1,35 @@
 import { useMemo } from 'react'
 
-import { type CategoryCounts, type CategoryId } from '@/types/categoryTypes'
 import { type NextServerSearchParams } from '@/types/searchParams'
-import { type Object } from '@/types/utils'
 
 import { CATEGORY_KEY } from '@/constants/searchParams'
 
 import { normalizeQueryParam } from '@/utils/queryUtils'
 
-export type UseCategoryProps<Entry extends Object> = {
-  searchParams: NextServerSearchParams
-  entries: Array<Entry>
-  validCategoryIds: Array<CategoryId>
+type EntryWithCategory = {
+  category: string
 }
```

Comment:
Why do we need this?

Reply:
This is more accurate (and safer) than `Object`.

We are calling `entry.category` below, so the entries we pass need a category key.

Reply:
As discussed, let's go with `HasCategory` or `WithCategory`.